library/common/engine.cc

```diff
@@ -21,17 +21,21 @@ Engine::Engine(envoy_engine_callbacks callbacks, const char* config, const char*
   main_thread_ = std::thread(&Engine::run, this, std::string(config), std::string(log_level));
 }
 
-envoy_status_t Engine::run(std::string config, std::string log_level) {
+envoy_status_t Engine::run(const std::string config, const std::string log_level) {
   {
     Thread::LockGuard lock(mutex_);
     try {
-      char* envoy_argv[] = {strdup("envoy"), strdup("--config-yaml"),   strdup(config.c_str()),
-                            strdup("-l"),    strdup(log_level.c_str()), nullptr};
+      const std::string name = "envoy";
+      const std::string config_flag = "--config-yaml";
+      const std::string log_flag = "-l";
+      const char* envoy_argv[] = {name.c_str(),     config_flag.c_str(), config.c_str(),
+                                  log_flag.c_str(), log_level.c_str(),   nullptr};
```

Comment:
Can you provide some motivation here?

Reply:
asan correctly complained about the engine test. the old strdup based implementation was leaking all these strings.

Reply:
We were leaking them intentionally before because we weren't sure how long Envoy needed them to stick around. This is fine though; they're on the stack anyways for the full duration of `run`.